src/main/java/hudson/plugins/ec2/util/EC2SlaveConfig.java

```diff
@@ -0,0 +1,285 @@
+package hudson.plugins.ec2.util;
+
+import hudson.model.Node;
+import hudson.plugins.ec2.AMITypeData;
+import hudson.plugins.ec2.ConnectionStrategy;
+import hudson.plugins.ec2.EC2Tag;
+import hudson.slaves.NodeProperty;
+
+import java.util.List;
+
+public abstract class EC2SlaveConfig {
```

Comment:
I'm just think out loud here, but this config with builder class pattern being developed here seems like a nice candidate for improving Jenkins' databinding feature to natively support this type of pattern.

Reply:
I'm generally of fan of builder patterns as it shows the user of the class a very verbose representation of what's used in constructing the class. `withSomeProperty()` is a lot more easier to understand at a quick glance than Constructor parameter #5 :)

That being said, I don't have that much experience with Jenkins' databinding, or Jenkins at all (other than using it) so I don't feel qualified to judge if this pattern works everywhere else :)

Reply:
Yeah, I don't, either, though I do have extensive experience in the equivalent functionality in Log4j2, so it's an active interest of mine in general. :)

The gist of my idea at least would be using `@DataBoundSetter` on the `with` or `set` methods and a `build()` method (or a new annotation), though I'm not sure how that would fit in with Stapler and Jenkins APIs. Hopefully I'll remember to revisit this after figuring that out.

Reply:
Haha yeah there's always the issue of remembering stuff.